File: workmanager/src/main/java/com/android/jetpack/workmanager/MainActivity.java
package com.android.jetpack.workmanager;

import android.annotation.TargetApi;
import android.app.AppOpsManager;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.content.pm.ApplicationInfo;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.os.SystemClock;
import android.provider.Settings;
import android.view.View;
import android.widget.Toast;

import com.android.jetpack.workmanager.jobs.DemoJobCreator;
import com.evernote.android.job.JobManager;

import java.lang.reflect.Field;
import java.lang.reflect.Method;
import java.text.SimpleDateFormat;
import java.util.Date;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NotificationCompat;
import androidx.work.WorkManager;

/**
 * Android-Job
 * @author zhangshuai
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.acBtn_startJob).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DemoJobCreator.scheduleJob(MainActivity.this);
            }
        });
        findViewById(R.id.acBtn_cancelJob).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DemoJobCreator.cancelByJobId(2);
            }
        });
    }
}

File: remotecallback/src/main/java/zs/jetpack/remotecallback/MainActivity.kt
package zs.jetpack.remotecallback

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * RemoteCallback 是一个用于在 Android 应用程序中处理远程回调的类。
 * 它提供了一种机制，使您可以在应用程序的不同组件之间进行跨进程通信，并处理远程方法调用的结果。
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}